### PR TITLE
Dependency: Update rules_go, Go, Python, curl, bazel-toolchains, bazelisk

### PR DESCRIPTION
**Pull request by moderation**
Signed-off-by: Michael Payne <michael@sooper.org>

Description: Update dependencies:
- rules_go 0.19.2 ([release](https://github.com/bazelbuild/rules_go/releases/tag/0.19.2)), [0.19.0 release notes](https://github.com/bazelbuild/rules_go/releases/tag/0.19.0). The big change is that `org_golang_google_grpc` is no longer declared in `go_rules_dependencies`. In order for the API to build, changes to `WORKSPACE` outlined [here](https://github.com/bazelbuild/rules_go/blob/0.19.0/go/workspace.rst#grpc-dependencies) are required. This pollutes `WORKSPACE` with some dependencies. Open on ways to improve this.
- Go 1.12.8 (to match rules_go)
- various Python updates
- curl 7.65.3 ([changelog](https://curl.haxx.se/changes.html#7_65_3))
- bazel-toolchains switches to 0.28.5 release and address @lizan TODO ([changes](https://github.com/bazelbuild/bazel-toolchains/compare/5a8611ee011d0d68498b16bf42a9c69d139bc708...0.28.5))
- bazelisk 1.0 ([release notes](https://github.com/bazelbuild/bazelisk/releases/tag/v1.0))

Risk Level: Low
Testing: `./docs/build.sh`, `bazel build @envoy_api//envoy/...`, `bazel test //test/...`
Docs Changes: None required
Release Notes: None required

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -15,3 +15,29 @@ envoy_dependencies()
 load("//bazel:dependency_imports.bzl", "envoy_dependency_imports")
 
 envoy_dependency_imports()
+
+load("@bazel_gazelle//:deps.bzl", "gazelle_dependencies", "go_repository")
```
Comment (lizan) [inline WORKSPACE]: roll them into `envoy_dependency_imports`.

Comment (moderation) [reply WORKSPACE]: @lizan I think I've done what you were suggesting. Local test, api build and format check look OK. Let me know if you were thinking of something different.

**Comment by moderation** [comment]
CI showing `envoy-macos` as Queued but it has run successfully and is duplicated further down in the checks. Go 1.12.9 was just released as was rules_go 0.19.3. Le sigh. Will bump in a follow up.

**Comment by lizan** [comment]
FWIW: macos CI passed but GH doesn't reflect https://dev.azure.com/cncf/envoy/_build/results?buildId=8456

**Pull request merged**